# Perguntas a serem respondidas 

## Qual foi o dia da semana com mais acidentes?
## E por estado?
## Qual foi a maior causa de acidentes?
## Qual o estado com o maior número de acidentes?
## Qual foi tipo de pista com o maior número de acidentes? E por estado?
## Responda essas perguntas para cada ano e o total da média dos anos

### Importação e limpeza de dados

In [238]:
import pandas as pd

In [239]:
df = pd.read_csv('datatrams/datatran2007.csv', encoding='latin', sep=';')

/home/italozdkj/.local/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3441: DtypeWarning: Columns (5,6) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [240]:
df.head()

id data_inversa dia_semana   horario  uf   br     km  municipio  \
0       10   11/06/2007    Segunda  15:30:00  MG  381  623.2   OLIVEIRA   
1       10   11/06/2007    Segunda  15:30:00  MG  381  623.2   OLIVEIRA   
2  1032898   13/08/2007    Segunda  14:25:00  MG   40  585.5  ITABIRITO   
3  1051130   12/02/2007    Segunda  02:10:00  MA  135   11.0   SAO LUIS   
4  1066824   20/11/2007      Terça  05:30:00  CE  222   30.8    CAUCAIA   

                causa_acidente            tipo_acidente  ... uso_solo   ano  \
0             Falta de atenção          Colisão frontal  ...    Rural  2007   
1             Falta de atenção          Colisão frontal  ...    Rural  2007   
2                       Outras           Saída de Pista  ...    Rural  2007   
3             Animais na Pista  Atropelamento de animal  ...   Urbano  2007   
4  Defeito mecânico em veículo              Capotamento  ...    Rural  2007   

  pessoas mortos feridos_leves feridos_graves ilesos  ignorados  feridos  \
0       9      0             4              0      5          0        4   
1       9      0             4              0      5          0        4   
2       3      0             0              1      2          0        1   
3       5      2             2              1      0          0        3   
4       1      0             1              0      0          0        1   

   veiculos  
0         6  
1         6  
2         1  
3         1  
4         1  

[5 rows x 26 columns]

In [241]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 127675 entries, 0 to 127674
Data columns (total 26 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   id                      127675 non-null  int64 
 1   data_inversa            127675 non-null  object
 2   dia_semana              127675 non-null  object
 3   horario                 127675 non-null  object
 4   uf                      127675 non-null  object
 5   br                      127675 non-null  object
 6   km                      127675 non-null  object
 7   municipio               127675 non-null  object
 8   causa_acidente          127675 non-null  object
 9   tipo_acidente           127675 non-null  object
 10  classificacao_acidente  127675 non-null  object
 11  fase_dia                127675 non-null  object
 12  sentido_via             127675 non-null  object
 13  condicao_metereologica  127675 non-null  object
 14  tipo_pista              127675 non-n

In [242]:
df.isna().sum()

id                        0
data_inversa              0
dia_semana                0
horario                   0
uf                        0
br                        0
km                        0
municipio                 0
causa_acidente            0
tipo_acidente             0
classificacao_acidente    0
fase_dia                  0
sentido_via               0
condicao_metereologica    0
tipo_pista                0
tracado_via               0
uso_solo                  0
ano                       0
pessoas                   0
mortos                    0
feridos_leves             0
feridos_graves            0
ilesos                    0
ignorados                 0
feridos                   0
veiculos                  0
dtype: int64

In [243]:
df.describe()

id       ano        pessoas         mortos  feridos_leves  \
count  1.276750e+05  127675.0  127675.000000  127675.000000  127675.000000   
mean   2.570106e+05    2007.0       2.101570       0.052806       0.452187   
std    6.161403e+05       0.0       1.562166       0.290575       1.081261   
min    1.000000e+01    2007.0       1.000000       0.000000       0.000000   
25%    2.138025e+05    2007.0       1.000000       0.000000       0.000000   
50%    2.529030e+05    2007.0       2.000000       0.000000       0.000000   
75%    2.898695e+05    2007.0       2.000000       0.000000       1.000000   
max    8.338491e+07    2007.0      80.000000      14.000000      52.000000   

       feridos_graves         ilesos      ignorados        feridos  \
count   127675.000000  127675.000000  127675.000000  127675.000000   
mean         0.184641       1.365201       0.046595       0.636828   
std          0.599220       1.084725       0.278431       1.335401   
min          0.000000       0.000000       0.000000       0.000000   
25%          0.000000       1.000000       0.000000       0.000000   
50%          0.000000       1.000000       0.000000       0.000000   
75%          0.000000       2.000000       0.000000       1.000000   
max         34.000000      51.000000      40.000000      62.000000   

            veiculos  
count  127675.000000  
mean        1.687104  
std         0.697045  
min         1.000000  
25%         1.000000  
50%         2.000000  
75%         2.000000  
max        19.000000

In [244]:
df.columns

Index(['id', 'data_inversa', 'dia_semana', 'horario', 'uf', 'br', 'km',
       'municipio', 'causa_acidente', 'tipo_acidente',
       'classificacao_acidente', 'fase_dia', 'sentido_via',
       'condicao_metereologica', 'tipo_pista', 'tracado_via', 'uso_solo',
       'ano', 'pessoas', 'mortos', 'feridos_leves', 'feridos_graves', 'ilesos',
       'ignorados', 'feridos', 'veiculos'],
      dtype='object')

In [245]:
df['uf'].unique()

array(['MG', 'MA', 'CE', 'PR', 'ES', 'GO', 'RJ', 'RS', 'SP', 'RN', 'SC',
       'PA', 'PE', 'MT', 'BA', 'AL', 'TO', 'PI', 'MS', 'SE', 'RO', 'PB',
       'AP', 'DF', 'AC', 'RR', 'AM', '(null)'], dtype=object)

In [246]:
df = df[df['uf'] != '(null)']

In [247]:
df = df[df['fase_dia']!='(null)']

In [248]:
df

id data_inversa dia_semana   horario  uf   br     km  \
0            10   11/06/2007    Segunda  15:30:00  MG  381  623.2   
1            10   11/06/2007    Segunda  15:30:00  MG  381  623.2   
2       1032898   13/08/2007    Segunda  14:25:00  MG   40  585.5   
3       1051130   12/02/2007    Segunda  02:10:00  MA  135   11.0   
4       1066824   20/11/2007      Terça  05:30:00  CE  222   30.8   
...         ...          ...        ...       ...  ..  ...    ...   
127670   956310   15/12/2007     Sábado  08:00:00  MG   50   22.0   
127671   964123   16/06/2007     Sábado  14:50:00  PR  373  268.3   
127672   985245   11/12/2007      Terça  14:00:00  MT   70  697.1   
127673   987413   09/05/2007     Quarta  14:50:00  SC  470   65.8   
127674   988556   07/02/2007     Quarta  07:15:00  BA  324  603.5   

            municipio               causa_acidente            tipo_acidente  \
0            OLIVEIRA             Falta de atenção          Colisão frontal   
1            OLIVEIRA             Falta de atenção          Colisão frontal   
2           ITABIRITO                       Outras           Saída de Pista   
3            SAO LUIS             Animais na Pista  Atropelamento de animal   
4             CAUCAIA  Defeito mecânico em veículo              Capotamento   
...               ...                          ...                      ...   
127670       ARAGUARI             Falta de atenção  Colisão com objeto fixo   
127671  PRUDENTOPOLIS                       Outras          Danos Eventuais   
127672        CACERES                       Outras          Colisão frontal   
127673        INDAIAL             Falta de atenção      Colisão Transversal   
127674   SIMOES FILHO             Falta de atenção         Colisão traseira   

        ... uso_solo   ano pessoas mortos feridos_leves feridos_graves ilesos  \
0       ...    Rural  2007       9      0             4              0      5   
1       ...    Rural  2007       9      0             4              0      5   
2       ...    Rural  2007       3      0             0              1      2   
3       ...   Urbano  2007       5      2             2              1      0   
4       ...    Rural  2007       1      0             1              0      0   
...     ...      ...   ...     ...    ...           ...            ...    ...   
127670  ...    Rural  2007       4      0             2              2      0   
127671  ...    Rural  2007       3      0             0              1      2   
127672  ...    Rural  2007       9      1             0              6      2   
127673  ...   Urbano  2007       3      0             3              0      0   
127674  ...   Urbano  2007       2      0             0              0      2   

        ignorados  feridos  veiculos  
0               0        4         6  
1               0        4         6  
2               0        1         1  
3               0        3         1  
4               0        1         1  
...           ...      ...       ...  
127670          0        4         1  
127671          0        1         2  
127672          0        6         3  
127673          0        3         2  
127674          0        0         2  

[127645 rows x 26 columns]

In [249]:
df['ocorrencia'] = df['id']

In [258]:
df.count()

id                        127645
data_inversa              127645
dia_semana                127645
horario                   127645
uf                        127645
br                        127645
km                        127645
municipio                 127645
causa_acidente            127645
tipo_acidente             127645
classificacao_acidente    127645
fase_dia                  127645
sentido_via               127645
condicao_metereologica    127645
tipo_pista                127645
tracado_via               127645
uso_solo                  127645
ano                       127645
pessoas                   127645
mortos                    127645
feridos_leves             127645
feridos_graves            127645
ilesos                    127645
ignorados                 127645
feridos                   127645
veiculos                  127645
ocorrencia                127645
dtype: int64

### Qual foi o dia da semana com mais acidentes?

In [250]:
df.groupby(by='dia_semana').count()['ocorrencia']

dia_semana
Domingo    20296
Quarta     16277
Quinta     17196
Segunda    16924
Sexta      20575
Sábado     20462
Terça      15915
Name: ocorrencia, dtype: int64

In [251]:
dia_semana = pd.DataFrame(df.groupby(by='dia_semana').count()['ocorrencia'])
dia_semana.sort_values(by='ocorrencia', ascending=False)

ocorrencia
dia_semana            
Sexta            20575
Sábado           20462
Domingo          20296
Quinta           17196
Segunda          16924
Quarta           16277
Terça            15915

####  O dia da semana com mais acidentes em todo o Brasil foi sexta-feira com 20575

### Qual foi o estado com maior número de acidentes?

In [266]:
df.groupby(by=['uf']).count()['ocorrencia']

uf
AC      324
AL     1604
AM      173
AP      150
BA     7086
CE     2260
DF     1138
ES     6242
GO     5607
MA     2112
MG    20600
MS     2299
MT     2636
PA     3127
PB     2695
PE     4762
PI     1881
PR     7073
RJ    11669
RN     3067
RO     2082
RR      458
RS    11037
SC    14584
SE     1159
SP    11109
TO      711
Name: ocorrencia, dtype: int64

In [314]:
df_estado = pd.DataFrame(df.groupby(by=['uf']).count()['id'].sort_values(ascending=False))

In [315]:
df_estado.columns

Index(['id'], dtype='object')

In [316]:
df_estado.head(15)

id
uf       
MG  20600
SC  14584
RJ  11669
SP  11109
RS  11037
BA   7086
PR   7073
ES   6242
GO   5607
PE   4762
PA   3127
RN   3067
PB   2695
MT   2636
MS   2299

#### O estado de MG foi o com a maior ocorrência de acidentes, com 20600 ocorrências.

### Causa de acidentes

In [325]:
df.groupby(by='causa_acidente').count()['id'].sort_values(ascending=False)

causa_acidente
Outras                                45962
Falta de atenção                      42247
Não guardar distância de segurança     8135
Velocidade incompatível                6005
Defeito mecânico em veículo            5403
Animais na Pista                       4201
Desobediência à sinalização            4041
Dormindo                               3363
Ingestão de álcool                     3103
Ultrapassagem indevida                 2837
Defeito na via                         2348
Name: id, dtype: int64

In [327]:
df_causas = pd.DataFrame(df.groupby(by='causa_acidente').count()['id'].sort_values(ascending=False))

In [329]:
df_causas

id
causa_acidente                           
Outras                              45962
Falta de atenção                    42247
Não guardar distância de segurança   8135
Velocidade incompatível              6005
Defeito mecânico em veículo          5403
Animais na Pista                     4201
Desobediência à sinalização          4041
Dormindo                             3363
Ingestão de álcool                   3103
Ultrapassagem indevida               2837
Defeito na via                       2348

#### A maior causa de acidentes foi falta de atenção, com 42247 registros.

In [382]:
uf = input(str('Digite o estado que deseja checar seu dia da semana mais perigoso: '))
por_estado_mais = df[df['uf']== uf.upper()].groupby(by='dia_semana').count()['id'].sort_values(ascending=False)
print(f'{por_estado_mais.head(1)}')

Digite o estado que deseja checar seu dia da semana mais perigoso: mt
dia_semana
Sábado    421
Name: id, dtype: int64


SyntaxError: invalid syntax (2805796963.py, line 2)

In [344]:
estados = pd.Series(df['uf'].unique())

In [346]:
estados.sort_values()

24    AC
15    AL
26    AM
22    AP
14    BA
2     CE
23    DF
4     ES
5     GO
1     MA
0     MG
18    MS
13    MT
11    PA
21    PB
12    PE
17    PI
3     PR
6     RJ
9     RN
20    RO
25    RR
7     RS
10    SC
19    SE
8     SP
16    TO
dtype: object